### PR TITLE
Small refactoring of TilingSprite

### PR DESCRIPTION
**Pull request by ivanpopelyshev**
Small refactoring of TilingSprite, allows to use renderer parts external plugins (pixi-picture)

### CODE REVIEW

**Comment by ivanpopelyshev** [comment]
Its not serious, I just need one review for that

File: src/extras/webgl/TilingSpriteRenderer.js [hunk 1]
```diff
@@ -94,9 +75,41 @@ export class TilingSpriteRenderer extends core.ObjectRenderer {
...
 
-        renderer.bindShader(shader);
+    /**
+     * For extra pixi plugins use
+     *
+     * @param {PIXI.TilingSprite} ts tiling sprite
+     * @param {boolean} isSimple is it repeating texture or not
+     * @param {PIXI.Shader} shader, must be set accordingly to isSimple
+     */
+    renderWithShader(ts, isSimple, shader)
```
Comment (englercj) [inline src/extras/webgl/TilingSpriteRenderer.js]: Why is this necessary over just setting the `.shader` property and calling `.render()` ?

File: src/extras/webgl/TilingSpriteRenderer.js [hunk 2]
```diff
@@ -130,17 +143,36 @@ export class TilingSpriteRenderer extends core.ObjectRenderer {
...
 
         const color = tempArray;
+        const alpha = ts.worldAlpha;
 
         core.utils.hex2rgb(ts.tint, color);
-        color[3] = ts.worldAlpha;
+        color[0] *= alpha;
+        color[1] *= alpha;
+        color[2] *= alpha;
+        color[3] = alpha;
```
Comment (englercj) [inline src/extras/webgl/TilingSpriteRenderer.js]: What is the benefit of multiplying here instead of in the fragment shader?

File: src/extras/webgl/TilingSpriteRenderer.js (hunk 1 above)
Comment (ivanpopelyshev) [inline src/extras/webgl/TilingSpriteRenderer.js]: so external code can set shader and blendMode and call that function.

**Comment by ivanpopelyshev** [comment]
The thing is, I have that code in separate plugin: https://github.com/pixijs/pixi-picture/blob/master/src/PictureRenderer.js#L184 , and I really don't want to copy existing TilingSprite rendering code there. I hope to move entire library into PIXI a bit later ;)

File: src/extras/webgl/TilingSpriteRenderer.js (hunk 2 above)
Comment (englercj) [inline src/extras/webgl/TilingSpriteRenderer.js]: Well as far as I can tell this is the only place where we multiply in code and not in the shader, so I would say this breaks the current convention.

File: src/extras/webgl/TilingSpriteRenderer.js (hunk 1 above)
Comment (ivanpopelyshev) [inline src/extras/webgl/TilingSpriteRenderer.js]: because blendMode that needs to be set must be different from sprite's blendMode. Of course I can just store it temporarily in other variable...

also shader needs to be bound before that to set some custom uniforms

Comment (ivanpopelyshev) [inline src/extras/webgl/TilingSpriteRenderer.js]: Also, which shader to choose depends on isSimple variable.

File: src/extras/webgl/TilingSpriteRenderer.js (hunk 2 above)
Comment (ivanpopelyshev) [inline src/extras/webgl/TilingSpriteRenderer.js]: But SpriteRenderer does that in vertex shader, which is different. I can o it there too, passing it to varying. Do you think that will be better?

File: src/extras/webgl/TilingSpriteRenderer.js (hunk 1 above)
Comment (ivanpopelyshev) [inline src/extras/webgl/TilingSpriteRenderer.js]: Actually, we cant just set ANY shader to TilingSprite and Sprite. What if we make some container for uniforms too, like we do with filter?

**Comment by ivanpopelyshev** [comment]
OK, screw this PR , I will make new one later
